Missing xgboost/lib/libxgboost.so

Following the [README#development](https://github.com/mljs/xgboost#development), 

If I 
```
$ git clone --recursive https://github.com/mljs/xgboost
$ cd xgboost
$ npm install                     // (Would be nice to include this also in the Development Section)
$ npm run build
```
gives the following error:

```
...
...
a - build/tree/updater_skmaker.o
a - build/tree/updater_sync.o
ERROR:root:dmlc-core/libdmlc.a: No such file or directory ("dmlc-core/libdmlc.a" was expected to be an input file, based on the commandline arguments provided)
ERROR:root:dmlc-core/libdmlc.a: No such file or directory ("dmlc-core/libdmlc.a" was expected to be an input file, based on the commandline arguments provided)
make[1]: *** [lib/libxgboost.dylib] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [xgboost] Error 1
fatal error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ar: fatal error in /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib
make[1]: *** [lib/libxgboost.a] Error 1
mkdir -p dist/wasm;
em++ -O3 -Wall -fPIC --memory-init-file 0 -std=c++11 -Ixgboost/dmlc-core/include -Ixgboost/rabit/include -Ixgboost/include js-interfaces.cpp xgboost/lib/libxgboost.so -o dist/wasm/xgboost.js --pre-js src/wasmPreJS.js -s WASM=1 -s "BINARYEN_METHOD='native-wasm'" -s ALLOW_MEMORY_GROWTH=1 -s EXPORTED_FUNCTIONS="['_create_model', '_set_param', '_train_full_model', '_predict_one', '_free_memory_model', '_save_model', '_get_file_content', '_load_model', '_prediction_size']"
ERROR:root:xgboost/lib/libxgboost.so: No such file or directory ("xgboost/lib/libxgboost.so" was expected to be an input file, based on the commandline arguments provided)
make: *** [build] Error 1
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! ml-xgboost@1.1.2 build: `rimraf dist && make clean && make`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the ml-xgboost@1.1.2 build script.
...
...
```

Important line I read was:

```
ERROR:root:xgboost/lib/libxgboost.so: No such file or directory ("xgboost/lib/libxgboost.so" was expected to be an input file, based on the commandline arguments provided)
```
When I checked the repo, it is missing the `lib` directory. A mention of https://github.com/dmlc/xgboost/blob/master/doc/build.md would be good to be included.

--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,13 @@ CFLAGS = -O3 -Wall -fPIC --memory-init-file 0 -std=c++11
 CFLAGS += -I$(DMLC_CORE)/include -I$(RABIT)/include -Ixgboost/include
 BUILD_DIR=dist
 EXPORTED_FUNCTIONS="['_create_model', '_set_param', '_train_full_model', '_predict_one', '_free_memory_model', '_save_model', '_get_file_content', '_load_model', '_prediction_size']"
-COMPILED_FILES = xgboost/lib/libxgboost.so
+
+UNAME=$(shell uname)
+ifeq ($(UNAME), Darwin)
+	COMPILED_FILES = xgboost/lib/libxgboost.dylib
+else
+	COMPILED_FILES = xgboost/lib/libxgboost.so
+endif
 
 all: replace build
 

